j2k/src/org/jetbrains/kotlin/j2k/OverloadReducer.kt

```diff
@@ -188,7 +188,14 @@ fun Converter.convertParameterList(
                 val parameter = params[i]
                 val defaultValue = overloadReducer?.parameterDefault(method, i)
                 val defaultValueConverted = if (defaultValue != null)
-                    deferredElement { codeConverter -> codeConverter.correctCodeConverter().convertExpression(defaultValue, parameter.type) }
+                    deferredElement { codeConverter ->
+                        val expr = codeConverter.correctCodeConverter().convertExpression(defaultValue, parameter.type)
+                        val refExpr = (defaultValue as? PsiReferenceExpression)
```

Comment:
It will not work when there are some operators around the expression.
```
boolean isA();
void foo() {
 foo(!isA())
}
void foo(boolean isA);
```

And when property used not in default value for clash position
```
boolean isA();
void foo() {
 foo(false, isA())
}
void foo(boolean isA, boolean isB);
```

Reply:
Please fix it and add to test case 🙂

Reply:
Fixed